dts/bindings/timer/st,stm32-timers.yaml

```
@@ -46,3 +46,12 @@ properties:
                                                down.
 
       If absent, then STM32_TIM_COUNTERMODE_UP is used (reset state).
+
+  st,deadtime:
+    type: int
+    default: 0
+    description: |
+      Sets the dead-time configuration in the Break and Dead-Time register (BDTR)
+      for the associated timer to prevent shoot-through currents in bridge
+      circuits.
```

Register name could vary across series.
```suggestion
      Sets the dead-time configuration to prevent shoot-through currents in bridge circuits.
```
Btw, some more detailed information be useful (unit), nature of the delay .
See for instance comment taken from Cube API (`Set the dead-time delay (delay inserted between the rising edge of the OCxREF signal and the rising edge of the Ocx and OCxN signals).)`